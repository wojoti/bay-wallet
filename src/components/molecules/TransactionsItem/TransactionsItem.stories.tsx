import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {TransactionsItem} from 'components/molecules';

export default {
  title: 'Molecules/TransactionsItem',
  component: TransactionsItem,
} as ComponentMeta<typeof TransactionsItem>;
const Template: ComponentStory<typeof TransactionsItem> = args => (
  <TransactionsItem {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
  expense: true,
  category: 'Withdraw',
  clientname: 'ATM',
  value: '-$1,000',
  date: '3d ago',
};
