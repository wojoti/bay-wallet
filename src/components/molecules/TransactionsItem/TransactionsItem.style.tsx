import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  wrapper: {
    marginVertical: 17,
    paddingHorizontal: 30,
  },
  logobox: {
    backgroundColor: '#346EC5',
    borderRadius: 8,
    width: 52,
    height: 52,
    justifyContent: 'center',
    alignItems: 'center',
    marginRight: 15,
  },
  detailsSpacer: {width: 'auto'},
});
export default style;
