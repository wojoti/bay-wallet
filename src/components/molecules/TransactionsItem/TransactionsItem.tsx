import {Container, Header, Img, ImgType, Label} from 'components/atoms';
import {PropsWithChildren} from 'react';
import {green, grey, marketDarkGrey, red} from 'theme/colors';
import style from './TransactionsItem.style';

export type TransactionsItemProps = PropsWithChildren<{
  expense: boolean;
  category: string;
  clientname: string;
  value: string;
  date: string;
  testId?: string;
}>;

const TransactionsItem = ({
  expense,
  category,
  clientname,
  value,
  date,
  testId,
}: TransactionsItemProps) => {
  const detailColor = expense ? red : green;
  return (
    <Container
      testId={testId}
      direction="row"
      align="center"
      justify="space-between"
      customStyle={style.wrapper}>
      <Container flex={4} direction="row" align="center">
        <Container customStyle={style.logobox}>
          <Img
            src={ImgType.bitcoin}
            height="21"
            alt="bitcoin logo"
            testId="transactionsitem-img"
          />
        </Container>
        <Container justify="center" align="flex-start">
          <Label
            color={marketDarkGrey}
            size={14}
            testId="transactionsitem-category">
            {category}
          </Label>
          <Header color="black" size={16} testId="transactionsitem-clientname">
            {clientname}
          </Header>
        </Container>
      </Container>
      <Container flex={1} justify="center" align="flex-end">
        <Header size={16} color={detailColor} testId="transactionsitem-value">
          {value}
        </Header>
        <Label size={14} color={grey} testId="transactionsitem-date">
          {date}
        </Label>
      </Container>
    </Container>
  );
};

export default TransactionsItem;
