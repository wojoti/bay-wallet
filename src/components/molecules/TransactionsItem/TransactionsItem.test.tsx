import '@testing-library/jest-native/extend-expect';
import {render, screen} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import TransactionsItem, {TransactionsItemProps} from './TransactionsItem';

const props: TransactionsItemProps = {
  testId: 'test-transactionsitem-id',
  expense: true,
  category: 'Withdraw',
  clientname: 'ATM',
  value: '-$1,000',
  date: '3d ago',
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<TransactionsItem {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render TransactionsItem', () => {
  render(<TransactionsItem {...props} />);
  const TransactionsItemElement = screen.getByTestId(
    'test-transactionsitem-id',
  );
  expect(TransactionsItemElement).toBeOnTheScreen();
});
