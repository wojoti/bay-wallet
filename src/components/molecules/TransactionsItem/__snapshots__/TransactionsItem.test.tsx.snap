// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      {
        "marginVertical": 17,
        "paddingHorizontal": 30,
      },
    ]
  }
  testID="test-transactionsitem-id"
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 4,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "stretch",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "width": "100%",
          },
          {
            "alignItems": "center",
            "backgroundColor": "#346EC5",
            "borderRadius": 8,
            "height": 52,
            "justifyContent": "center",
            "marginRight": 15,
            "width": 52,
          },
        ]
      }
    >
      <SvgMock
        height="21"
        testID="transactionsitem-img"
        width="100%"
      />
    </View>
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "flex": undefined,
            "flexDirection": "column",
            "justifyContent": "center",
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          {
            "color": "#3D454B",
            "fontFamily": "Poppins-Light",
            "fontSize": 14,
            "fontWeight": "400",
            "textAlign": "left",
          }
        }
        testID="transactionsitem-category"
      >
        Withdraw
      </Text>
      <Text
        style={
          {
            "color": "black",
            "fontFamily": "Poppins-Bold",
            "fontSize": 16,
            "textAlign": "left",
          }
        }
        testID="transactionsitem-clientname"
      >
        ATM
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "flex-end",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        {
          "color": "#E52020",
          "fontFamily": "Poppins-Bold",
          "fontSize": 16,
          "textAlign": "left",
        }
      }
      testID="transactionsitem-value"
    >
      -$1,000
    </Text>
    <Text
      style={
        {
          "color": "#6B6B6B",
          "fontFamily": "Poppins-Light",
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        }
      }
      testID="transactionsitem-date"
    >
      3d ago
    </Text>
  </View>
</View>
`;
